server/api/section/section.controller.js

```diff
@@ -59,6 +61,30 @@ function removeEntity(res) {
   };
 }
 
+function addStudentSection(section, pendingStudent, callback){
+  var asyncTasks = [];
+  asyncTasks.push((callback)=>{
+    section.students.push(pendingStudent);
+    section.save(()=>{
+      callback();
+    });
+  });
+
+  asyncTasks.push(()=>{
+    User.findByIdAsync(pendingStudent)
+    .then( (student) => {
+      student.sections.push(section._id);
+      student.save( () =>{
+        callback();
```

Comment:
It's also worth noting that this is an unsafe operation ([see two phase commit](https://docs.mongodb.org/v2.4/tutorial/perform-two-phase-commits/)). What if there's an error when saving the section? The student will have a sections and the section won't have the student. The opposite is true as well.